charts/spire/charts/spire-server/templates/roles.yaml

```diff
@@ -3,7 +3,7 @@
 kind: Role
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: {{ include "spire-server.fullname" . }}
+  name: {{ include "spire-server.fullname" . }}-pods
```

Comment:
Why changing this role name? Arn't roles per definition used for pods?

Reply:
The previous role/rolebinding is split into two because the role/rolebinding only provides permissions into the namespace the role/rolebinding is created in. In this case, the ability to deal with the configmap is in the other namespace so needs separate objects.

Roles can impart permissions to all object types, not just pods.

Reply:
Actually, I think there is a cleaner way to do this now that you make me look at it a little closer.....

the pod permissions in the rule are actually dead code. Its only able to look up pods in the local namespace with it. Its the cluster role /binding later in the file that actually lets it get the pods cluster wide.

Reply:
Except, now that upstream certmanager is a thing, that still needs to be in the spire-server namespace....

Will have to rework this a bit.

Reply:
K. reworked it to fix the dead pod code and incorporate the merged cert-manager code that merged kind of poorly.